### ISSUE TITLE
Trigger soft pull to activate lasers

### ISSUE THREAD

**Issue by shiftyscales** [Frooxius/NeosPublic#486]
Discussed earlier on the Discord, a user's lasers could be visible to them only during a partial trigger press. Upon release of the trigger, the laser will disappear after ~1s regardless of if it is targetting an object.

This change prevents "sticky lasers", and also leaves the trigger from accidentally activating buttons, etc. that the laser is attached to during the click.

**Comment by Frooxius** [comment]
I'm not sure I understand what do you mean by "only during partial trigger press". This could be added  for comfort, but I don't see how this helps with sticky lasers or trigger accidentally activating buttons, can you elaborate on that?

**Comment by shiftyscales** [comment]
Yes. So right now, in order to activate/de-activate the lasers, it requires that the user fully depress the trigger until it clicks. Because lasers currently don't disappear on their own while they are targetting something interactable, e.g. buttons, trying to hide the laser will result in a click.

My proposal is instead of the laser's on/off state being a toggle, it is instead a 'hold' action. By tying the laser's on state to a soft-pull, it frees the actual click input for deliberate clicks, and by being a hold action, it deactivates the laser when the user is done with it (has released their trigger) rather than needing to wait until their laser is free from the range of interactible things.

This desire also allows more flexibility in how a user could configure their laser, e.g. having a high stick threshold without compromising their ability to turn off their lasers when they're done using them, they'd instead just release the trigger instead of needing to fully depress it to a click while their laser isn't targetting anything.

**Comment by shiftyscales** [comment]
To further elaborate on this issue based on feedback by GearBell- it takes a long time to activate the new lasers because it requires a full trigger click every couple of seconds, then a second click to actually peform the action. Click to activate laser -> Grip to drag the Logix node around. Click to activate laser -> Click to activate the button.

And the underlying issue resulting in that- is because the lasers can sometimes be too quick to time out.

An issue I see with the timeout based approach is that the user's desired timeout can vary greatly by workflow. E.g. if they're just pressing a button once, they might not want their laser to stay on for 5+ seconds afterward, but if they're working consistently with Logix, etc. they will be regularly using their laser/might not want it to time out at all.

A soft-pull to activate the laser allows the user to perfrom the desired action on the first click, and for the laser to not be visible any longer than it has to be to perform that action.

**Comment by TehTurk** [comment]
So to add to this, the reason for the request on the partial press to activate lasers, is the ability to just have the lasers work automatically turning off and on as they had before when interacting with certain objects, (Tooltips, grabbables, anchors etc)

**Comment by Frooxius** [comment]
I've added this in 2020.5.8.141. It feels pretty good to use and has some benefits too. I'm closing this for now, but reopen if there are issues or changes and such.

**Comment by shiftyscales** [comment]
- The soft-pull feels as good as I imagined.

- The laser timeout (in seconds) chould be adjustable.

- The above request can wait until the setting UI is reworked so that you don't have to double your workload by implementing it twice.

- A timeout of 0 could instantly dismiss the laser once the trigger is released below the threshold for activation.

- Others could set their timeout according to whatever feels comfortable for them as well.

- The default timeout could be brought down considerably from where it is in the current build. Perhaps one second? Users can already keep their laser always-on if they hold the trigger past the threshold.

- The current default timeout is also refreshed by the laser hovering over interactible objects- so the laser stays active longer than it should due to the default timeout being too long as noted above.

- (If it's trivial to implement, a setting in the Config.json to adjust the timeout would be greatly appreciated by me in the short-term until it can get implemented proper.)

Thanks for implementing this Froox. <3
